### PR TITLE
HARVESTER: Add external links to detail/masthead

### PR DESCRIPTION
**Pull request by n313893254**
### Summary
- [Add external links to detail/masthead](https://github.com/harvester/dashboard/pull/523)
- [Harvester: Fix detail page external link](https://github.com/harvester/dashboard/pull/570)

### CODE REVIEW

File: shell/components/ResourceDetail/Masthead.vue [hunk 1]
```diff
@@ -366,6 +366,10 @@ export default {
     hideNamespaceLocation() {
       return this.$store.getters['currentProduct'].hideNamespaceLocation;
     },
+
+    resourceExternalLink() {
+      return this.value.resourceExternalLink;
```
Comment (mantis-toboggan-md) [inline shell/components/ResourceDetail/Masthead.vue:371]: Where is this computed prop used?

**Comment by mantis-toboggan-md** [comment]
A fix for the empty name bug has been merged here: https://github.com/rancher/dashboard/pull/7941

File: shell/components/ResourceDetail/Masthead.vue (hunk 1 above)
Comment (n313893254) [reply shell/components/ResourceDetail/Masthead.vue:371]: Done. Thanks for your review.

**Comment by mantis-toboggan-md** [comment]
Hi @n313893254 sorry for the delay in re-reviewing this. This PR is specifically to allow adding a link to the Longhorn UI for PVCs right? This is a very specific use-case to modify a very general component; I'm not sure how many other places would use `resourceExternalLink`. 

The recent work by @aalves08 would allow the Harvester plugin to add information to detail/masthead for targeted resource types within the Harvester product. See examples here: https://github.com/aalves08/dashboard/tree/0264f1ed6fc173073740cef2498b51711373f76e/pkg/test-features


For this use, it would look like:
```
  plugin.addPanel(
    PanelLocation.DETAILS_MASTHEAD,
    { resource: ['persistentvolumeclaim'] },
    { component: () => import('../components/ResourceExternalLink.vue') });
 ```
 Where ResourceExternalLink.vue would be:
 ```
 <script>
export default {
  name:  'ResourceExternalLink',
  props: {
    resource: {
      type:    Object,
      default: () => {
        return {};
      }
    }
  }
};
</script>

<template>
  <div>
    <a
      v-if="resource.resourceExternalLink"
      :href="resourceExternalLink.url"
    >
      View in Longhorn UI 
      <i class="icon icon-external-link" />
    </a>
  </div>
</template>
```
 The component is rendered below the resource name:
 
Screen Shot 2023-02-10 at 11 57 48 AM


Please let me know if this will not work for the Harvester issue.

**Comment by n313893254** [comment]
Hi, @DaiYuzeng. Could you please help validate it?

**Comment by n313893254** [comment]
Closed in favor of Yuzeng validate https://github.com/rancher/dashboard/pull/8025#issuecomment-1426217072 work. Harvester UI will use the feature after the next rebase.

image

**Pull request closed without merging**